### PR TITLE
Print proxy type as an operator

### LINKED ISSUE #3076: Issue with pretty-printing nested proxies
If I have a nested proxy: `@(@"Hi")` it pretty-prints as `@@"Hi"` which isn't valid syntax:

```
Unable to parse module:
  unexpected @@
  expecting type
```

### PR DESCRIPTION
**Pull request by LiamGoodacre**
Fixes #3076 

Example input:
```purescript
x :: _
x = {x: @(@(@(Array Int))), x: @Int}
```
Output:
> Label x appears more than once in a row type.
> while inferring the type of
> ```purescript
> { x: @(@(@(Array Int)))
> , x: @Int
> }
> ```

I originally also changed the parser so it treated `@` as a prefix type operator that binds tighter than type application, but this caused problems with instances because `@Int` isn't a type-atom.  E.g: `instance foo :: Foo @t` would have to have been written as `instance foo :: Foo (@t)`.  So I reverted this.

### CODE REVIEW

File: src/Language/PureScript/Pretty/Types.hs [hunk 1]
```diff
@@ -74,6 +74,12 @@ prettyPrintRowWith tro open close = uncurry listToBox . toList []
 prettyPrintRow :: Type -> String
 prettyPrintRow = render . prettyPrintRowWith defaultOptions '(' ')'
 
+proxyType :: Pattern () Type (Type, Type)
+proxyType = mkPattern match
+  where
+  match (ProxyType t) = Just (TypeConstructor (Qualified Nothing (ProperName "@")), t)
```
Comment (paf31) [inline src/Language/PureScript/Pretty/Types.hs]: Interesting trick :)

Comment (paf31) [reply src/Language/PureScript/Pretty/Types.hs]: I'm wondering if we should add `Prefix` to `pattern-arrows` going forward, but I'm fine with this in the meantime. Could you please add a comment though?

**Comment by LiamGoodacre** [comment]
Rebased and commented.  Let me know if you want it to say something else.

**Comment by paf31** [comment]
👍 Sounds good, thanks!

**Pull request merged**